lib/rules/no-unsupported-features/es-syntax.js

```diff
@@ -435,10 +435,15 @@ function defineVisitor(context, options) {
      * @returns {boolean} `true` if it's supporting.
      */
     function isNotSupportingVersion(aCase) {
-        return (
-            !aCase.supported ||
-            options.version.intersects(getSemverRange(`<${aCase.supported}`))
-        )
+        if (!aCase.supported) {
+            return true
+        }
+
+        if (aCase.supported instanceof Range) {
+            return !options.version.intersects(aCase.supported)
+        }
+
+        return options.version.intersects(getSemverRange(`<${aCase.supported}`))
```

Comment:
This _could_ be written as a single Boolean expression, but I figured expanding it was a bit more clear.